### PR TITLE
DO NOT MERGE: upgrade to go 1.12

### PR DESCRIPTION
**Pull request by rboyer**
Filing this PR so that CI can run the tests.

At the very least these now seem to fail:

* `go test ./agent -run TestHTTPAPI_Ban_Nonprintable_Characters`
* `go test ./command/lock -run TestLockCommand_BadArgs`

### CODE REVIEW

**Comment by rboyer** [comment]
`TestLockCommand_BadArgs` is failing because of a small `flag.DurationVar` error message string change:
```
// 1.11.5
func (d *durationValue) Set(s string) error {
    v, err := time.ParseDuration(s)
    *d = durationValue(v)
    return err
}
```

```
// 1.12
func (d *durationValue) Set(s string) error {
    v, err := time.ParseDuration(s)
    if err != nil {
        err = errParse
    }
    *d = durationValue(v)
    return err
}
```
Our tests test for the presence of the very string that is now different.

**Comment by rboyer** [comment]
The `TestHTTPAPI_Ban_Nonprintable_Characters` failure seems to be related to this change in `net/http`: https://github.com/golang/go/commit/f1d662f34788f4a5f087581d0951cdf4e0f6e708

Our unit tests assume in these scenarios `http.NewRequest` does not return an error and this is now false. Correcting this may be made more interesting due to this user setting: https://www.consul.io/docs/agent/options.html#disable_http_unprintable_char_filter

**Comment by rboyer** [comment]
I changed the tests to no longer swallow the `http.NewRequest` error (instead asserting no error) and now it fails properly instead of straight segfaulting:
```
Exit code: 1
--- FAIL: TestHTTPAPI_Ban_Nonprintable_Characters (0.44s)
    http_test.go:329: 
        	Error Trace:	http_test.go:329
        	Error:      	Received unexpected error:
        	            	parse /v1/kv/badness: net/url: invalid control character in URL
        	Test:       	TestHTTPAPI_Ban_Nonprintable_Characters
--- FAIL: TestHTTPAPI_Allow_Nonprintable_Characters_With_Flag (0.34s)
    http_test.go:342: 
        	Error Trace:	http_test.go:342
        	Error:      	Received unexpected error:
        	            	parse /v1/kv/badness: net/url: invalid control character in URL
        	Test:       	TestHTTPAPI_Allow_Nonprintable_Characters_With_Flag
Binary file test.log matches
```

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ language: go
 go:
   # Please keep this in-sync with the go version we build against in
   # build-support/docker/Build-Go.dockerfile.
```
Comment (banks) [inline .travis.yml]: Wonder if we should add this comment to the circle CI config too?

**Pull request closed without merging**